cfme/web_ui/utilization.py

```diff
@@ -0,0 +1,790 @@
+            return final_data
+        # MiddlewareServer
+        elif self.id == 'jvm_non_heap_usage_bytes' and page_type == "MiddlewareServer":
+            raw_used = mgmt.metric.list_gauge_server(
+                feed_id=feed_id, server_id=server_name, bucket_duration=bucket_duration,
+                metric_enum=MetricEnumGauge.SVR_MEM_NON_HEAP_USED, start=start, end=end)
+            raw_committed = mgmt.metric.list_gauge_server(
+                feed_id=feed_id, server_id=server_name, bucket_duration=bucket_duration,
+                metric_enum=MetricEnumGauge.SVR_MEM_NON_HEAP_COMMITTED, start=start, end=end)
+            # merge all data
+            final_data = self._update_mgmt_data(key='used', raw_data=raw_used)
+            final_data = self._update_mgmt_data(key='committed', raw_data=raw_committed,
+                                                pre_final_data=final_data)
+            return final_data
+        # MiddlewareServer
+        elif self.id == 'gc_duration_ms' and page_type == "MiddlewareServer":
+            raw_duration = mgmt.metric.list_counter_server(
+                feed_id=feed_id, server_id=server_name, bucket_duration=bucket_duration,
+                metric_enum=MetricEnumCounter.SVR_MEM_ACCUMULATED_GC_DURATION,
+                start=start, end=end)
+            return self._update_mgmt_data(key='duration', raw_data=raw_duration)
+        # MiddlewareServer
+        elif self.id == 'web_sessions' and page_type == "MiddlewareServer":
+            raw_active = mgmt.metric.list_gauge_server(
+                feed_id=feed_id, server_id=server_name, bucket_duration=bucket_duration,
+                metric_enum=MetricEnumGauge.SVR_WEB_AGGREGATED_ACTIVE_WEB_SESSIONS,
+                start=start, end=end)
+            raw_expired = mgmt.metric.list_counter_server(
+                feed_id=feed_id, server_id=server_name, bucket_duration=bucket_duration,
+                metric_enum=MetricEnumCounter.SVR_WEB_AGGREGATED_EXPIRED_WEB_SESSIONS,
+                start=start, end=end)
+            raw_rejected = mgmt.metric.list_counter_server(
+                feed_id=feed_id, server_id=server_name, bucket_duration=bucket_duration,
+                metric_enum=MetricEnumCounter.SVR_WEB_AGGREGATED_REJECTED_WEB_SESSIONS,
+                start=start, end=end)
+            final_data = self._update_mgmt_data(key='active', raw_data=raw_active)
+            final_data = self._update_mgmt_data(key='expired', raw_data=raw_expired,
+                                                pre_final_data=final_data)
+            final_data = self._update_mgmt_data(key='rejected', raw_data=raw_rejected,
+                                                pre_final_data=final_data)
+            return final_data
+        # MiddlewareServer
+        elif self.id == 'transactions' and page_type == "MiddlewareServer":
+            raw_committed = mgmt.metric.list_counter_server(
```

Comment:
Consider using partial for raw since only 1 parameter is varied here.

```python
from functools import partial

raw = partial(mgmt.metric.list_counter_server, feed_id=feed_id, [...other consistent params...])

raw_committed = raw(metric_enum=MetricEnumCounter.THING_COMMITTED)